### PR TITLE
feat(optimizer): activate rules

### PR DESCRIPTION
**Pull request by MingjiHan99**
Signed-off-by: MingjiHan <mingjihan@umass.edu>

### CODE REVIEW

File: src/optimizer/mod.rs [hunk 1]
```diff
@@ -32,7 +32,14 @@ impl Optimizer {
         // plan = plan.rewrite(&mut ConstantFolding);
         // plan = plan.rewrite(&mut ArithExprSimplification);
         // plan = plan.rewrite(&mut BoolExprSimplification);
```
Comment (st1page) [inline src/optimizer/mod.rs:34]:
```suggestion
```

**Review by skyzh — APPROVED** [review]
LGTM. Would you please fix fmt and (if possible) add test cases for the rewriters?

File: src/optimizer/logical_plan_rewriter/arith_expr_simplification.rs [hunk 2]
```diff
@@ -62,3 +62,34 @@ impl Rewriter for ArithExprSimplification {
         *expr = new;
     }
 }
+
+impl PlanRewriter for ArithExprSimplificationRule {
+    fn rewrite_logical_join(&mut self, join: &LogicalJoin) -> PlanRef {
+        let left = self.rewrite(join.left());
+        let right = self.rewrite(join.right());
+        Arc::new(join.clone_with_rewrite_expr(left, right, self))
+    }
```
Comment (wangrunji0408) [inline src/optimizer/logical_plan_rewriter/arith_expr_simplification.rs:71]: I don't think it's elegant to manually implement these functions for pure expression rules.
Is there a way to resolve that? cc @st1page

Comment (st1page) [reply src/optimizer/logical_plan_rewriter/arith_expr_simplification.rs:71]: maybe we could implement a kind of `PlanRewriter`, give it a Box<ExprRewriter> when initalizing it. all Expression rewrite demand will use the same code.

Comment (st1page) [reply src/optimizer/logical_plan_rewriter/arith_expr_simplification.rs:71]: or another way, how about doing expression simplificatation when construct the expression? so we can make sure any expression appearing in our project has been simplificated. just a half-baked idea.

Comment (st1page) [reply src/optimizer/logical_plan_rewriter/arith_expr_simplification.rs:71]: Anyway, we can refactor in future PRs.

Comment (wangrunji0408) [reply src/optimizer/logical_plan_rewriter/arith_expr_simplification.rs:71]: > or another way, how about doing expression simplificatation when construct the expression? so we can make sure any expression appearing in our project has been simplificated. just a half-baked idea.

This way we will lose the original expressions, not good for debugging. -1 (

**Comment by MingjiHan99** [comment]
> LGTM. Would you please fix fmt and (if possible) add test cases for the rewriters?

We will add them in the incoming PR.

**Pull request merged**